### ISSUE TITLE
Addition Of Dolby AC-4 to YT-DLP

### ISSUE THREAD

**Issue by Karthik-20034** [yt-dlp/yt-dlp#5738]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Example URLs

https://www.hotstar.com/in/movies/cadaver/1260109485/watch

### Provide a description that is worded well enough to be understood

Hey Developers,

> when I came across to download the audio codec of a video.it shows normally m4a audio,EAC-3 audio.But not supporting the latest Audio Codec, Dolby AC-4.so it’s my humble request to add this audio codec support for sites.

> Reference image for the feature is given below.
![2F6D16F1-0FCB-453A-B0E6-ABB3762A5C87](https://user-images.githubusercontent.com/83129674/206098687-6d6eadd4-c8b6-4e34-9465-103830987975.jpeg)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[~/Documents]$ yt-dlp https://www.hotstar.com/in/movies/cadaver/1260109485/watch -f sdr-dash-audio_1666342630095_ta_ac-4-0-N 2 -r 6200
000 -vU --cookies ~/Documents/cookies/hotstar.com_cookies.txt
[debug] Command-line config: ['https://www.hotstar.com/in/movies/cadaver/1260109485/watch', -f', 'sdr-dash-audio_1666342630095_ta_ac-4
-0', '-N', '2',
'-r', '6200000', '-vU',
'--cookies', /private/var/mobile/Containers/Data/Application/BD2A5D8E-25C8-44FB-900D-2C265568A
D32/Documents/cookies/hotstar.com_cookies.txt']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8 [debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.9.10+ (Python iPad12,1 64bit) - macOS-16.1.1-iPad12,1-arm-64bit (OpenSSL 1.1.1i 8 Dec 2020)
[debug] exe versions: ffmpeg n5.0.1-4-gaebb3d25e (setts), ffprobe n5.0.1-4-gaebb3d25e, phantoms present, rtmpdump present [debug] Optional libraries: Cryptodome-3.15.0, certifi-2022.06.15, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4 [debug] Proxy map: {} [debug] Loaded 1723 extractors [debug] Fetching release info: https://api.github.com/repos/y-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11 yt-dlp is up to date (2022.11.11)
[debug] [hotstar] Extracting URL: https://www.hotstar.com/in/movies/cadaver/1260109485/watch
[hotstar] 1260109485: Downloading JSON metadata [hotstar] 1260109485: Downloading webpage [hotstar] 1260109485: Downloading JSON metadata [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading MPD manifest
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading MPD manifest
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading m3u8 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading MPD manifest
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading mu8 information
WARNING: Unknown codec ac-4.02.02.00
```

**Comment by pukkandan** [comment]
For devs: The codec just needs to be added to `FormatSorter`, `get_compatible_ext` and `parse_codecs`

**Comment by Karthik-20034** [comment]
Can You inform when the codec is added to YT-DLP.I am excited to test the codec..

**Comment by Karthik-20034** [comment]
I noticed that it will not download with audio and does not merge with video after post processing.can you help with that.

**Comment by pukkandan** [comment]
> I noticed that it will not download audio with video and does not merge with video after post processing.can you help with that.
> 
> >

Show verbose log of this

**Comment by Karthik-20034** [comment]
[~/Documents$ yt-dlp https://www.hotstar.com/in/movies/cadaver/1260109485/watch - sdr-dash-video_h265_4k_dr_1660528186967-hvc1-3-2+s dr-dash-audio [1666342630095](tel:1666342630095) ta ac-4-4 -N 2 -r 720000 -vU --cookies ~/Documents/cookies/hotstar.com cookies.txt [debug] Command-line config: ['https://www.hotstar.com/in/movies/cadaver/1260109485/watch', '-f', 'sdr-dash-video_h265_4k_sdr_166052818
6967-hvc1-3-2+sdr-dash-audio_1666342630095_ta_ac-4-4','-N', '2', '-r', '720000',
"-VU'
'-cookies', "/private/var/mobile/Containers/D
ata/Application/BD2A5D8E-25C8-44FB-900D-2C265568AD32/Documents/cookies/hotstar.comcookies.txt']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8 [debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.9.10+ (Python iPad12,1 64bit) - macOS-16.1.1-iPad12,1-arm-64bit (OpenSSL 1.1.1i 8 Dec 2020)
[debug] exe versions: ffmpeg n5.0.1-4-gaebb3d25e (setts), ffprobe n5.0.1-4-gaebb3d25e, phantoms present, rtmpdump present [debug] Optional libraries: Cryptodome-3.15.0, certifi-2022.06.15, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4 [debug] Proxy map: {} debuql Loaded 1723 extractors [debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11 yt-dlp is up to date (2022.11.11)
[debug] [hotstar] Extracting URL: https://www.hotstar.com/in/movies/cadaver/1260109485/watch
[hotstar] [1260109485](tel:1260109485): Downloading JSON metadata [hotstar] [1260109485](tel:1260109485): Downloading webpage [hotstar] [1260109485](tel:1260109485): Downloading JSON metadata [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest [hotstar] [1260109485](tel:1260109485): Downloading m38 information [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest [hotstar] [1260109485](tel:1260109485): Downloading mu8 information [hotstar] [1260109485](tel:1260109485): Downloading m3u8 information
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00 [hotstar] [1260109485](tel:1260109485): Downloading mu8 information [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest
WARNING: Unknown codec ac-4.02.02.00 [hotstar] [1260109485](tel:1260109485): Downloading m38 information [hotstar] [1260109485](tel:1260109485): Downloading m38 information [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest [hotstar] [1260109485](tel:1260109485): Downloading MPD manifest

**Comment by Karthik-20034** [comment]
It will Download the highest quality video,but not download Dolby AC-4 audio.

**Comment by pukkandan** [comment]
Send me account details and I'll check

Contact: discord (pukkandan#4207) / reddit (/u/pukkandan) / email (pukkandan.ytdlp@gmail.com)

**Comment by pukkandan** [comment]
Sure

**Comment by Karthik-20034** [comment]
I’ve sent the email.Please kindly check it.

**Comment by Karthik-20034** [comment]
Any Updates ??

**Comment by Karthik-20034** [comment]
Any Updates ?

**Comment by Karthik-20034** [comment]
I Just want to know that is this bug/error fixed through an update? 
Or will these patched with this current version.

**Comment by pukkandan** [comment]
Neither of the provided cookies work
```
❯ yt-dlp hotstar.com/in/movies/cadaver/1260109485/watch --cookies hotstar.com_cookies_1.txt
WARNING: [generic] The url doesn't specify the protocol, trying with http
[hotstar] 1260109485: Downloading JSON metadata
[hotstar] 1260109485: Downloading webpage
[hotstar] 1260109485: Downloading token
[hotstar] 1260109485: Downloading JSON metadata
ERROR: [hotstar] 1260109485: Unable to download JSON metadata: HTTP Error 402: Payment Required (caused by <HTTPError 402: 'Payment Required'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

❯ yt-dlp hotstar.com/in/movies/cadaver/1260109485/watch --cookies hotstar.com_cookies.txt
WARNING: [generic] The url doesn't specify the protocol, trying with http
[hotstar] 1260109485: Downloading JSON metadata
[hotstar] 1260109485: Downloading webpage
[hotstar] 1260109485: Downloading token
[hotstar] 1260109485: Downloading JSON metadata
ERROR: [hotstar] 1260109485: Unable to download JSON metadata: HTTP Error 402: Payment Required (caused by <HTTPError 402: 'Payment Required'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

**Comment by Karthik-20034** [comment]
Please check the email once more and work around immediately as soon as possible.the cookies will revoke every 18hrs.

**Comment by Karthik-20034** [comment]
Please check your email,if anything happens,please comment me.

**Comment by pukkandan** [comment]
The cookies are likely bound to your IP. You may have to provide actual account details

```
❯ yt-dlp hotstar.com/in/movies/cadaver/1260109485/watch --cookies hotstar.com_cookies_2.txt
WARNING: [generic] The url doesn't specify the protocol, trying with http
[hotstar] 1260109485: Downloading JSON metadata
[hotstar] 1260109485: Downloading webpage
[hotstar] 1260109485: Downloading token
[hotstar] 1260109485: Downloading JSON metadata
ERROR: [hotstar] 1260109485: Unable to download JSON metadata: HTTP Error 402: Payment Required (caused by <HTTPError 402: 'Payment Required'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

**Comment by Karthik-20034** [comment]
It is logged in with my phone number.OTP will be provided.how will I do that you.

**Comment by Karthik-20034** [comment]
Have you looked into it , is it working now ?

**Comment by pukkandan** [comment]
> I forwarded the cookies file for the third time a minute ago.did you checked that ?

See above log



> It is logged in with my phone number.OTP will be provided.how will I do that you.



> Contact: discord (pukkandan#4207) / reddit (/u/pukkandan) / email ([pukkandan.ytdlp@gmail.com](mailto:pukkandan.ytdlp@gmail.com))

**Comment by Karthik-20034** [comment]
I ve sent a message to your discord.please respond to it.

**Comment by Karthik-20034** [comment]
[~/Documents]$ yt-dlp https://www.hotstar.com/in/movies/cadaver/1260109485/watch - sdr-dash-video_h265_4k_sdr_1660528186967-hv1-3-2+sd r-dash-audio_1666342630095_ta_ac-4-0 -N 2 -r 6200000 -vU --cookies ~/Documents/cookies/hotstar.com_cookies.txt [debug] Command-line config: ['https://www.hotstar.com/in/movies/cadaver/1260109485/watch', '-f',
'sdr-dash-video_h265_4k_sdr_166052818
6967-hv1-3-2+sdr-dash-audio_1666342630095_ta_ac-4-0','-N', '2',
'-r', '6200000', '- VU',
--cookies', '/private/var/mobile/Containers/D
ata/Application/BD2A5D8E-25C8-44FB-900D-2C265568D32/Documents/cookies/hotstar.com_cookies.txt']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8 [debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.9.10+ (Python iPad12,1 64bit) - macOS-16.1.1-iPad12,1-arm-64bit (OpenSSL1.1.li 8 Dec 2020)
[debug] exe versions: ffmpeg n5.0.1-4-gaebb3d25e (setts), ffprobe n5.0.1-4-gaebb3d25e, phantoms present, rtmpdump present [debug] Optional libraries: Cryptodome-3.15.0, certifi-2022.06.15, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4 [debug] Proxy map: {} [debug] Loaded 1723 extractors [debug] Fetching release info: https://api.qithub.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11 yt-dlp is up to date (2022.11.11)
[debug] [hotstar] Extracting URL: https://www.hotstar.com/in/movies/cadaver/1260109485/watch
[hotstar] 1260109485: Downloading JSON metadata [hotstar] 1260109485: Downloading webpage [hotstar] 1260109485: Downloading JSON metadata [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading m3u8 information
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading m38 information
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading m3u8 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest
[hotstar] 1260109485: Downloading mu8 information
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading m3u8 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading mu8 information
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading m3u8 information [hotstar] 1260109485: Downloading m38 information
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading m38 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading m3u8 information [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading mu8 information
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading MPD manifest [hotstar] 1260109485: Downloading MPD manifest
WARNING: Unknown codec ac-4.02.02.00 [hotstar] 1260109485: Downloading MPD manifest
WARNING: Unknown codec ac-4.02.02.00
[hotstar] 1260109485: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx
, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: [hotstar] 1260109485: Requested format is not available. Use --list-formats for a list of available formats
Traceback (most recent call last):
  File "/var/mobile/Containers/Data/Application/BD2A5D8E-25C8-44FB-900D-2C265568AD32/Library/lib/python3.9/site-packages/yt_dlp/Youtube
DL.py", line 1485, in wrapper
    return func(self, *args, **kwargs)
  File "/var/mobile/Containers/Data/Application/BD2A5D8E-25C8-44FB-900D-2C265568AD32/Library/lib/python3.9/site-packages/yt_dlp/Youtube
DL.py", line 1582, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/var/mobile/Containers/Data/Application/BD2A5D8E-25C8-44FB-900D-2C265568AD32/Library/lib/python3.9/site-packages/yt_dlp/Youtube
DL.py", line 1641, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/var/mobile/Containers/Data/Application/BD2A5D8E-25C8-44FB-900D-2C265568AD32/Library/lib/python3.9/site-packages/yt_dlp/Youtube
DL.py", line 2701, in process_video_result
    raise ExtractorError(
yt_dlp.utils.ExtractorError: [hotstar] 1260109485: Requested format is not available. Use --list-formats for a list of available formats

**Comment by pukkandan** [comment]
There are 3 distinct issues here:
1. yt-dlp doesnt recognize ac4 and so shows warning - fixed by 0bbddc92f76e3040327115a75e869f1fb71e675c
2. ffmpeg does not support this codec. It raises a PostprocessingError during merge. We'll just have to wait for ffmpeg to add support (ref: https://trac.ffmpeg.org/ticket/8349)
3. You are passing wrong format id. No clue where u r getting this id... You need to fix that urself
